homeassistant/components/jewish_calendar/binary_sensor.py

```diff
@@ -66,14 +68,22 @@ class JewishCalendarBinarySensorEntityDescription(
 
 async def async_setup_entry(
     hass: HomeAssistant,
-    config_entry: ConfigEntry,
+    config: ConfigEntry,
     async_add_entities: AddEntitiesCallback,
 ) -> None:
     """Set up the Jewish Calendar binary sensors."""
-    async_add_entities(
-        JewishCalendarBinarySensor(
-            hass.data[DOMAIN][config_entry.entry_id], description
+    entry = hass.data[DOMAIN][config.entry_id]
+    for description in BINARY_SENSORS:
+        async_update_unique_id(
+            hass,
+            config.entry_id,
+            entry["old_prefix"],
+            BINARY_SENSOR_DOMAIN,
+            description.key,
         )
+
```

Comment:
Honestly, just centralize this in `__init__.py`. This leaves you with less bloated code

Reply:
Thanks. Done.